src/Instrumentation/ReactHttp/src/ReactHttpInstrumentation.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\ReactHttp;
+
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Message\ResponseInterface;
+use React\Http\Io\Transaction;
+use React\Http\Message\ResponseException;
+use React\Promise\PromiseInterface;
+use Throwable;
+
+/** @psalm-suppress UnusedClass */
+class ReactHttpInstrumentation
+{
+    public const NAME = 'react-http';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.react-http',
+            schemaUrl: 'https://opentelemetry.io/schemas/1.32.0',
+        );
+
+        /** @psalm-suppress UnusedFunctionCall */
+        hook(
+            Transaction::class,
+            'send',
```

Comment:
This hook call is for the extension to intercept and do it's magic right?

Reply:
that's right, it registers the pre/post callbacks via the zend observer API, so that they are executed when that function is called.

Reply:
Is this the preferred way, even for maintainers, to implement things? Because I do like that this doesn't add any code to projects directly and keeps the instrumentation isolated from a package's logic.

Reply:
@WyriHaximus from https://opentelemetry.io/docs/specs/otel/overview/#instrumentation-libraries
> The inspiration of the project is to make every library and application observable out of the box by having them call OpenTelemetry API directly. However, many libraries will not have such integration, and as such there is a need for a separate library